gamestonk_terminal/cryptocurrency/cryptocurrency_helpers.py

```diff
@@ -197,145 +167,113 @@ def find(other_args: List[str]):
 
     Parameters
     ----------
-    other_args: List[str]
-        Arguments to pass to argparse
+    top: int
+        Number of records to display
+    coin: str
+        Cryptocurrency
+    key: str
+        Searching key (symbol, id, name)
+    source: str
+        Data source of coins.  CoinGecko (cg) or CoinPaprika (cp) or Binance (bin)
+    export : str
+        Export dataframe data to csv,json,xlsx file
     """
-    parser = argparse.ArgumentParser(
-        prog="find",
-        add_help=False,
-        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
-        description="""
-        Find similar coin by coin name,symbol or id. If you don't remember exact name or id of the Coin at CoinGecko,
-        Binance or CoinPaprika you can use this command to display coins with similar name, symbol or id
-        to your search query.
-        Example of usage: coin name is something like "polka". So I can try: find -c polka -k name -t 25
-        It will search for coin that has similar name to polka and display top 25 matches.
-        -c, --coin stands for coin - you provide here your search query
-        -k, --key it's a searching key. You can search by symbol, id or name of coin
-        -t, --top it displays top N number of records.""",
-    )
-    parser.add_argument(
-        "-c",
-        "--coin",
-        help="Symbol Name or Id of Coin",
-        dest="coin",
-        required="-h" not in other_args,
-        type=str,
-    )
-    parser.add_argument(
-        "-k",
-        "--key",
-        dest="key",
-        help="Specify by which column you would like to search: symbol, name, id",
-        type=str,
-        choices=["id", "symbol", "name"],
-        default="symbol",
-    )
-    parser.add_argument(
-        "-t",
-        "--top",
-        default=10,
-        dest="top",
-        help="Limit of records",
-        type=check_positive,
-    )
 
-    parser.add_argument(
-        "--source",
-        dest="source",
-        choices=["cp", "cg", "bin"],
-        default="cg",
-        help="Source of data.",
-        type=str,
-    )
-
-    try:
+    if source == "cg":
+        coins_df = get_coin_list()
+        coins_list = coins_df[key].to_list()
+        sim = difflib.get_close_matches(coin, coins_list, top)
+        df = pd.Series(sim).to_frame().reset_index()
+        df.columns = ["index", key]
+        coins_df.drop("index", axis=1, inplace=True)
+        df = df.merge(coins_df, on=key)
+        print(
+            tabulate(
+                df,
+                headers=df.columns,
+                floatfmt=".1f",
+                showindex=False,
+                tablefmt="fancy_grid",
+            )
+        )
+        export_data(
+            export,
+            os.path.dirname(os.path.abspath(__file__)),
+            "find",
+            df,
```

Comment:
,"\n"

Reply:
Here we don't need "\n", because there is `print("")` which is done after all tabulate print statements:

```
def find(source: str, coin: str, key: str, top: int, export: str) -> None:
    ...
    if source == "cg":
        ....
        print(
            tabulate(
                df,
                headers=df.columns,
                floatfmt=".1f",
                showindex=False,
                tablefmt="fancy_grid",
            )
        )
    elif source == "cp":
        ...
        print(
            tabulate(
                df,
                headers=df.columns,
                floatfmt=".1f",
                showindex=False,
                tablefmt="fancy_grid",
            )
        )

    elif source == "bin":
         ...
        print(
            tabulate(
                df,
                headers=df.columns,
                floatfmt=".1f",
                showindex=False,
                tablefmt="fancy_grid",
            )
        )
    else:
        print("Couldn't execute find methods for CoinPaprika, Binance or CoinGecko")
    print("")

```
So the last line `print("")` is executed at the end. It's done this way to not repeat 4  time print statement :D Something that Didier suggested.